pkg/api/api.go

```diff
@@ -19,6 +19,12 @@ import (
 	"golang.org/x/oauth2"
 )
 
+var querier database.Querier
```

Comment:
I agree with you and the post in most cases, because using package state/ global state can make code less flexible and less testable.
However, making a wrapper struct also has structural overhead that complicates the implementation. In our case, we must consider the nature that the package is the API layer, which is not a library that is supposed to have a programmatic client, i.e. no code will call the package. So we have to ask ourselves following questions before making the design decision about whether we should use package state here:

1) Do we need multiple implementation or instances of the api package that will be used in different scenarios? if yes, go for wrapper struct;

2) Do we need to use different implementations or instances for the fan-out of the package? for instance, for test purposes? if yes, go for wrapper struct;

My answer to the above question is 1) we only need one instance/implementation of the api package and 2) we will not make multiple instances of the fan-outs.
So to be pragmatic, I choose the simplest implementation of the api layer.
This is the most appropriate design as I can see. If the situation changes in the future, we can refactor it, but not necessarily now.

Also please consider why google allows package state / global state, if it is always "bad", they can simply remove the feature or not having it at first place:)